.github/workflows/pipeline.yml

```diff
@@ -106,6 +108,22 @@ jobs:
         id: plan
         run: terraform plan -input=false -out=plan
 
+      - name: Create new GitHub tag
+        if: github.ref == 'refs/heads/master'
+        id: new_tag
+        uses: butlerlogic/action-autotag@stable
+        with:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+          strategy: package
+          tag_prefix: "v"
+
+      - name: Check if tag was created
+        if: github.ref == 'refs/heads/master'
+        run: |
+          if [ "${{ steps.new_tag.outputs.tagcreated }}" != "yes" ]; then
+            exit 1
+          fi
```

Comment:
This step prevents pushes to master without bumping the version. The action before will automatically create tags for a new version and has an output that says whether or not a new tag was created (an output we use here).